### PR TITLE
Add website link checker

### PR DESCRIPTION
**Pull request by lucperkins**
This PR adds a link checker for the website. It can be run using `make check-links`. It assumes that the website has already been built, checks only internal links, and has *not* yet been added to the CI pipeline (it's currently intended only to be run locally).

The PR also includes some link fixes in `/docs/latest` and adds some missing image `alt`s.

### CODE REVIEW

File: docs/.htmltest.yml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+DirectoryPath: website/public
+IgnoreDirectoryMissingTrailingSlash: true
+CheckExternal: false
```
Comment (patrick-east) [inline docs/.htmltest.yml]: We should probably set `IgnoreAltMissing` or see if we can filter the results more.. as-is the output is kind of crazy with the versioned content.. its like `2937 errors in 267 documents` where almost all of them are missing alt tags in old content (same ones are repeated for each version).

File: docs/content/rest-api.md [hunk 2]
```diff
@@ -710,7 +710,7 @@ Content-Type: application/json
 
 ## Data API
 
-The Data API exposes endpoints for reading and writing documents in OPA. For an introduction to the different types of documents in OPA see [How Does OPA Work?](../#how-does-opa-work).
+The Data API exposes endpoints for reading and writing documents in OPA. For an introduction to the different types of documents in OPA see [How Does OPA Work?](../how-does-opa-work).
```
Comment (patrick-east) [inline docs/content/rest-api.md]: Funny enough, this change regresses on the preview:

```
{15:34} ~ ❯ docker run --rm -it linkchecker/linkchecker https://deploy-preview-2156--openpolicyagent.netlify.com/

<snip>

URL        `../how-does-opa-work'
Name       `How Does OPA Work?'
Parent URL https://deploy-preview-2156--openpolicyagent.netlify.com/docs/edge/rest-api/, line 2893, col 144
Real URL   https://deploy-preview-2156--openpolicyagent.netlify.com/docs/edge/how-does-opa-work
Check time 3.287 seconds
Size       1KB
Result     Error: 404 Not Found

<snip>
```
I think the crux of the issue is that there is no "how does opa work" section anymore. This should probably just be pointing to `../` (which the invalid `#` was inadvertently doing for us 😆 )

**Comment by patrick-east** [comment]
@lucperkins any update?

**Comment by lucperkins** [comment]
@patrick-east Oops, this slipped through the cracks. I'll have a go at this tomorrow.

**Comment by lucperkins** [comment]
I don't quite have the bandwidth for this at the moment, so I'm gonna close for now and re-open when I can devote some bandwidth to it

**Pull request closed without merging**